pymeasure/instruments/lakeshore/lakeshore421.py

```diff
@@ -0,0 +1,395 @@
+    .. code-block:: python
+
+        gaussmeter = LakeShore421("COM1")
+        gaussmeter.unit = "T"               # Set units to Tesla
+        gaussmeter.auto_range = True        # Turn on auto-range
+        gaussmeter.fast_mode = True         # Turn on fast-mode
+
+    """
+
+    MULTIPLIERS = {1e3: 'k', 1: '', 1e-3: 'm', 1e-6: 'n'}
+    RANGES = {0.003: 0, 0.03: 1, 0.3: 2, 3: 3}  # in Tesla
+    PROBETYPES = {"High Sensitivity": 0,
+                  "High Stability": 1,
+                  "Ultra-High Sensitivity": 2}
+
+    def __init__(self, resource_name, **kwargs):
+        super(LakeShore421, self).__init__(
+            resource_name,
+            "Lake Shore 421 Gaussmeter",
+            read_termination='\r',
+            write_termination='\n',
+            command_delay=0.05,
+            **kwargs
+        )
+
+        self.adapter.connection.set_visa_attribute(
+            visa_const.VI_ATTR_ASRL_BAUD, 9600)
+        self.adapter.connection.set_visa_attribute(
+            visa_const.VI_ATTR_ASRL_DATA_BITS, 7)
+        self.adapter.connection.set_visa_attribute(
+            visa_const.VI_ATTR_ASRL_STOP_BITS, visa_const.VI_ASRL_STOP_ONE)
+        self.adapter.connection.set_visa_attribute(
+            visa_const.VI_ATTR_ASRL_PARITY, visa_const.VI_ASRL_PAR_ODD)
+
+    def _raw_to_field(self, field_raw, multiplier_name):
+        if not field_raw == "OL":
+            multiplier = getattr(self, multiplier_name)
+            field = multiplier * field_raw
+        else:
+            field = np.nan
+
+        return field
+
+    def _field_to_raw(self, field, multiplier_name):
+        multiplier = getattr(self, multiplier_name)
+        return field / multiplier
+
+    field_raw = Instrument.measurement(
+        "FIELD?",
+        """ Returns the field in the current units and multiplier
+        """,
+    )
+
+    field_multiplier = Instrument.measurement(
+        "FIELDM?",
+        """ Returns the field multiplier for the returned magnetic field.
+        """,
+        values=MULTIPLIERS,
+        map_values=True
+    )
+
+    @property
+    def field(self):
+        """ Returns the field in the current units. This property takes into
+        account the field multiplier. Returns np.nan if field is out of range.
+        """
+        return self._raw_to_field(self.field_raw, "field_multiplier")
+
+    unit = Instrument.control(
+        "UNIT?", "UNIT %s",
+        """ A string property that controls the units used by the gaussmeter.
+        Valid values are G, T, Oe, A/m. """,
+        validator=strict_discrete_set,
+        values=['G', 'T', 'Oe', 'A/m'],
+    )
+
+    field_range = Instrument.control(
+        "RANGE?", "RANGE %d",
+        """ A floating point property that controls the field range of the
+        meter. Valid values are 0.003, 0.03, 0.3, and 3 (in Tesla). """,
+        validator=truncated_discrete_set,
+        values=RANGES,
+        map_values=True
```

Comment:
Generally, it's advisable to include a trailing comma in the last argument (or a dict entry). That way, when you add another argument, you don't have to change this line, too.

Reply:
Yes, you're right, I'd normally do this, but now I seemed to have forgotten.